code/game/objects/items/weapons/storage/bags.dm

```diff
@@ -115,6 +115,13 @@
 	can_hold = list(/obj/item/weapon/reagent_containers/food/snacks/grown,/obj/item/seeds,/obj/item/weapon/grown)
 	burn_state = FLAMMABLE
 
+/obj/item/weapon/storage/bag/plants/can_be_inserted(obj/item/W, stop_messages = 0, mob/living/user)
+	if(istype(W, /obj/item/weapon/reagent_containers/food/snacks/grown/cherry_bomb))

```

Comment:
In all seriousness that check is really snowflake and bad, I'm not sure what the ideal solution is though